ref: Remove SentryPrivate (#3623)
getsentry/sentry-cocoa: CHANGELOG.md

## Unreleased

### Improvements

> - Remove SentryPrivate from Carthage, self built, and CocoaPods (#3623)

Author: ```suggestion
- Remove SentryPrivate library (#3623)
```

### Improvements

- Cache installationID async to avoid file IO on the main thread when starting the SDK (#3601)

> - Remove SentryPrivate from Carthage and self build (#3623)

Author: ```suggestion
- Remove SentryPrivate from Carthage and self build and Cocoapod (#3623)
```

> - Remove SentryPrivate from Carthage and self build and Cocoapod (#3623)

Reviewer: `m`:
```suggestion
- Remove SentryPrivate from Carthage, self built, and CocoaPods (#3623)
```
